static/js/containers/DashboardPage_test.js

```diff
@@ -203,27 +209,51 @@ describe('DashboardPage', () => {
       );
     });
 
-    it('renders a price', () => {
-      let calculatePricesStub = helper.sandbox.stub(libCoupon, 'calculatePrices');
-      let dashboard = makeDashboard();
-      let availablePrograms = makeAvailablePrograms(dashboard);
-      let coursePrices = makeCoursePrices(dashboard);
+    describe('course pricing', () => {
+      let dashboard, availablePrograms, coursePrices, calculatePricesStub;
+      let run, program;
 
-      // set enrollment information so run comes up as offered in dashboard
-      let program = dashboard[0];
-      let run = program.courses[0].runs[0];
-      run.enrollment_start_date = '2016-01-01';
+      beforeEach(() => {
+        calculatePricesStub = helper.sandbox.stub(libCoupon, 'calculatePrices');
+        dashboard = makeDashboard();
+        program = dashboard[0];
+        run = program.courses[0].runs[0];
+        run.enrollment_start_date = '2016-01-01';
+        availablePrograms = makeAvailablePrograms(dashboard);
+        coursePrices = makeCoursePrices(dashboard);
+        helper.dashboardStub.returns(Promise.resolve(dashboard));
+        helper.programsGetStub.returns(Promise.resolve(availablePrograms));
+        helper.coursePricesStub.returns(Promise.resolve(coursePrices));
+      });
 
-      helper.dashboardStub.returns(Promise.resolve(dashboard));
-      helper.programsGetStub.returns(Promise.resolve(availablePrograms));
-      helper.coursePricesStub.returns(Promise.resolve(coursePrices));
-      let calculatedPrices = new Map([[run.id, 123]]);
-      calculatePricesStub.returns(calculatedPrices);
+      it('renders a price', () => {
+        let calculatedPrices = new Map([[run.id, 123]]);
+        calculatePricesStub.returns(calculatedPrices);
 
-      return renderComponent('/dashboard', DASHBOARD_SUCCESS_ACTIONS).then(([wrapper]) => {
-        assert.include(wrapper.text(), `Pay Now $123`);
+        return renderComponent('/dashboard', DASHBOARD_SUCCESS_ACTIONS).then(([wrapper]) => {
+          assert.include(wrapper.text(), `Pay Now $123`);
+          assert.isTrue(calculatePricesStub.calledWith(dashboard, coursePrices, []));
+        });
+      });
 
-        assert.isTrue(calculatePricesStub.calledWith(dashboard, coursePrices, []));
+      it('should issue a request to skip if all the course prices are 0', () => {
```

Comment:
Can we add a test that verifies that we don't try to skip if `has_user_applied` is already true?